service/internal_error.go

```diff
@@ -0,0 +1,24 @@
+package service
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	bitriseOAuthInternalError = "Bitrise OAuth internal error"
+)
+
+// InternalError ...
+type InternalError struct {
+	Err error
+}
+
+func (e *InternalError) Error() string {
+	return fmt.Sprintf("%s: %v", bitriseOAuthInternalError, e.Err)
+}
+
+// IsInternalError returns TRUE if the error is an internal error
+func IsInternalError(err error) bool {
+	return strings.Contains(err.Error(), bitriseOAuthInternalError)
+}
```

Comment:
Sorry, but I do not get it quite exactly. Would you mind unfolding a bit more this?
On `code` did you mean HTTP status code? How this could be distinct enough to unequivocally identify this type of error?
From my understanding using a `flag` would look similar to this solution, just maybe with type or something like that?

Reply:
Unfortunately, the new error type called `InternalError` returned by the round-tripper, is getting additionally wrapped by other errors as well, so a simple type check is not possible in the error writer to distinguish it, that is why I used this "workaround".

Reply:
I was thinking something like: `err = &InternalError{Err: err, Internal: true}` and then we could use this for checking instead of string matching. Error messages could change version by version so if we rely on that it would make harder to update change things. However a flag is explicit and remains the same no matter what.

Reply:
The error we get in the error writer is also wrapped by another error (the error emitted upon an invalid IP for example is wrapped into an `http.httpError` if I'm not mistaken), which is I'm not sure that will be the same for every type of issue. However to do a flag check, we would need the error, but if the `InternalError` would be easily accessible, a simple type check could be the solution. So currently due to not knowing all of the possible errors (and their wrapped combinations), the most straightforward solution seemed to be doing a string contains on the stringified error.